activemq-broker/src/main/java/org/apache/activemq/broker/region/Queue.java

```diff
@@ -1971,10 +1978,17 @@ private final boolean tryCursorAdd(final Message msg) throws Exception {
 
     final void messageSent(final ConnectionContext context, final Message msg) throws Exception {
         pendingSends.decrementAndGet();
+
         destinationStatistics.getEnqueues().increment();
         destinationStatistics.getMessages().increment();
         destinationStatistics.getMessageSize().addSize(msg.getSize());
 
+        MessageFlowStats tmpMessageFlowStats = destinationStatistics.getMessageFlowStats();
+
+        if(isAdvancedMessageStatisticsEnabled() && tmpMessageFlowStats != null) {
```

Comment:
This is something where both of these should ALWAYS be in sync. you should never have the stats enabled but the object is null.

Maybe the solution is you drop the flag entirely. Maybe stats being enabled is really just a "is this object not null" and disabling is making the stats null so we don't have to check 2 objects.